ariac_gazebo/ariac_gazebo/environment_startup.py

```diff
@@ -1116,10 +1118,12 @@ def fill_part_lot_msg(self, part: PartInfo, quantity: int):
     def spawn_conveyor_part(self):
         if self.conveyor_enabled:
             if self.conveyor_spawn_order == 'sequential':
-                part_params = self.conveyor_parts_to_spawn.pop(0)
+                part_params = next(self.conveyor_parts_to_spawn_cycle)
             elif self.conveyor_spawn_order == 'random':
-                part_params = self.conveyor_parts_to_spawn.pop(
-                    randint(0, len(self.conveyor_parts_to_spawn) - 1))
+                part_params = next(self.conveyor_parts_to_spawn_random_cycle)
+
+            part_params.name = part_params.name.replace(
+                part_params.name[part_params.name.find("_"):], "_" + (str(next(self.conveyor_loop_iter) % 10000)).zfill(4))
```

Comment:
The part names don't need to have a consistent number of digits. I know that we would never realistically get to 10,000 parts but I don't want to create a system with a set limit that is not necessary. I would just append `"_" + (str(next(self.conveyor_loop_iter)` to the name string.